language/move-vm/runtime/src/move_vm.rs

```diff
@@ -14,6 +14,17 @@ impl MoveVM {
         }
     }
 
+    /// Create a new Session backed by the given storage.
+    ///
+    /// Right now it is the caller's responsibility to ensure cache coherence of the Move VM Loader
+    ///   - In general, a new Move VM needs to be created whenever changes to the published code
```

Comment:
I don't see how publishing a Module requires a VM to be recreated. In fact, that should not be the case. Maybe I am reading this wrong, but I should be able to publish and keep going, as a matter of fact that is what we do today, so hopefully that is correct behavior

Reply:
Oops, looks like I generalized the rules a little bit too much (on the conservative side). Got it updated.